File: blog_vue/src/model/folder/FolderInformation.ts
import Folder from "./Folder";
import File from "./File";

export default class FolderInformation extends Folder {
  username: string;
  folderList: Array<Folder>;
  fileList: Array<File>;

  constructor() {
    super();
    this.username = "";
    this.folderList = [];
    this.fileList = [];
  }
}

File: blog_springBoot/src/main/java/com/synergism/blog/core/repository/entity/FolderInformation.java
package com.synergism.blog.core.repository.entity;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Setter
@Getter
public class FolderInformation extends Folder {

    private List<Folder> folderList;

    private List<File> fileList;
}
